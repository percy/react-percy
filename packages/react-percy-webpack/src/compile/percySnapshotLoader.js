const percyGlobals = [
  'after',
  'afterAll',
  'afterEach',
  'before',
  'beforeAll',
  'beforeEach',
  'percySnapshot',
  'suite',
];

const eslintWhitelistGlobals =
  '/* global ' + percyGlobals.map(global => `${global}:false`).join(', ') + ' */ ';

function percySnapshotLoader(source, map) {
  return this.callback(null, eslintWhitelistGlobals + source, map);
}

module.exports = percySnapshotLoader;
