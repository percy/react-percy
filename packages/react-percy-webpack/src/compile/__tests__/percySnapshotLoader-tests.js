import percySnapshotLoader from '../percySnapshotLoader';

it('prepends ESLint comment whitelisting Percy globals to source code', () => {
  const webpack = {
    callback: jest.fn(),
  };
  const source = `import foo from './bar';`;
  const map = { mock: 'source map' };

  percySnapshotLoader.call(webpack, source, map);

  const transformedSource = webpack.callback.mock.calls[0][1];
  expect(transformedSource).toMatchSnapshot();
});
