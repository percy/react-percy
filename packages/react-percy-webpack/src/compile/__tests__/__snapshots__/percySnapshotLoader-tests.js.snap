// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepends ESLint comment whitelisting Percy globals to source code 1`] = `"/* global after:false, afterAll:false, afterEach:false, before:false, beforeAll:false, beforeEach:false, percySnapshot:false, suite:false */ import foo from './bar';"`;
